### PR TITLE
Add seals environments and corresponding (tentative) hyperparameters.

### PR DESCRIPTION
**Pull request by ernestum**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 ### New Features
 - Support setting PyTorch's device with thye `--device` flag (@gregwar)
+- Add `--tital-n-trials` parameter to help with distributed optimization. (@ernestum)
```
Comment (AdamGleave) [inline CHANGELOG.md]:
```suggestion
- Add `--total-n-trials` parameter to help with distributed optimization. (@ernestum)
```

File: hyperparams/ppo.yml [hunk 2]
```diff
@@ -53,6 +70,24 @@ MountainCar-v0:
...
+  policy: 'MlpPolicy'
+  batch_size: 512
+  clip_range: 0.2
+  ent_coef: 6.4940755116195606e-06
+  gae_lambda: 0.98
+  gamma: 0.99
+  learning_rate: 0.0004476103728105138
+  max_grad_norm: 1
+  n_epochs: 20
+  n_steps: 256
```
Comment (AdamGleave) [inline hyperparams/ppo.yml:87]: Interesting that this is a lot larger than for `MountainCar-v0` (although that one was perhaps not tuned)

File: tests/test_train.py [hunk 3]
```diff
@@ -101,3 +103,47 @@ def test_parallel_train(tmp_path):
...
+        str(100),
+        "--algo",
+        "ppo",
+        "--env",
+        "Pendulum-v1",
+        "--log-folder",
+        tmp_path,
+    ]
+
+    p1 = subprocess.Popen(["python", "train.py"] + args)
```
Comment (AdamGleave) [inline tests/test_train.py]: Seems a bit easier to use list comprehension?
```
processes = [subprocess.Popen(["python", "train.py"] + args) for _ in range(4)]
assert all([p.wait() == 0 for p in processes])
```
etc

File: utils/exp_manager.py [hunk 4]
```diff
@@ -747,7 +751,26 @@ def hyperparameters_optimization(self) -> None:
...
 
         try:
-            study.optimize(self.objective, n_trials=self.n_trials, n_jobs=self.n_jobs)
+            if self.total_n_trials is not None:
+                counted_states = [
+                    TrialState.COMPLETE,
+                    TrialState.RUNNING,
+                    TrialState.PRUNED,
+                ]
+                completed_trials = sum(t.state in counted_states for t in study.trials)
```
Comment (AdamGleave) [inline utils/exp_manager.py]: Do we want to count something as `TrialState.RUNNING` as completed? (Genuine question, not familiar with Optuna.)

**Review by AdamGleave — COMMENTED** [review]
Mostly LGTM, left a few minor inline comments.

One high-level comment for the hyperparams files: when hyperparams are just copy-pasted from existing config, perhaps better to make use of the `&some_env_default`, `<<: *some_env_default` YAML syntax to make this more explicit?

**Comment by ernestum** [comment]
Thanks for the review @AdamGleave! This is a bit of a confusing PR, since it is constantly being rebased on top of https://github.com/DLR-RM/rl-baselines3-zoo/pull/240, which contains all the functionality that is going upstream. Most of your comments would rather apply to that PR.

File: utils/exp_manager.py (hunk 4 above)
Comment (ernestum) [reply utils/exp_manager.py]: We count them as "potentially completed". If we do not count already running trials, then we get (total_n_trials + n_workers) many trials. Which obviously is too much and also is confusing.
The drawback is, that if a worker crashes and its trial is left in a RUNNING state, we have to delete the stale trial manually.
See also https://github.com/optuna/optuna/issues/3518 for details.

**Pull request merged**